src/core/featuremodel.cpp

```diff
@@ -319,7 +319,15 @@ void FeatureModel::resetAttributes()
 
 void FeatureModel::applyGeometry()
 {
-  mFeature.setGeometry( mGeometry->asQgsGeometry() );
+  QgsGeometry geometry = mGeometry->asQgsGeometry();
+
+  const QList<QgsVectorLayer *> intersectionLayers = QgsProject::instance()->avoidIntersectionsLayers();
+  if ( !intersectionLayers.isEmpty() && geometry.type() == QgsWkbTypes::PolygonGeometry )
+  {
+    geometry.avoidIntersections( intersectionLayers );
```

Comment:
so it is working within a set of layers and strictly within one layer?
That sounds a bit awkward to me: let's say I have 2 layers: cities and regions. I have cities without regions yet. I am drawing the new region, I want to avoid overlap with existing but I would expect it to be on top of existing cities.
Does it make sense?

Reply:
Yes. So, there's a bigger talk to have here, which I'm keen to do with you and @m-kuhn this afternoon if possible.

This implementation matches the behavior of QGIS, which IMHO QField should do. That being said, I think QGIS' behavior is totally broken, and we need to improve things there. It goes back to what we chatted about on Monday: ATM, the avoid overlap is based on layers checked in the snapping advanced config. I think we need to add a tri-state overlap config, that'd be: allow overlap, disallow overlap on currently selected layer, respect advanced config overlap checks.

Reply:
> I think we need to add a tri-state overlap config

+1
I didn't understand that the current behavior was like this. I would go for this, yes.

Reply:
@3nids , OK, so further changes will be made on the QGIS front. I don't think it'll require changes on the QField side of things.